addons/edi_stock/models/edi_move_cancellation_report_document.py

```diff
@@ -0,0 +1,171 @@
+"""EDI stock move request records"""
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class StockMove(models.Model):
+    """Extend ``stock.move`` to include the EDI Stock Cancellation document"""
+
+    _inherit = "stock.move"
+
+    edi_stock_cancellation_doc_id = fields.Many2one(
+        "edi.document",
+        string="EDI Stock Cancellation Document",
+        required=False,
+        readonly=True,
+        index=True,
+    )
+
+
+class EdiDocument(models.Model):
+    """Extend ``edi.document`` to include stock move cancellation records"""
+
+    _inherit = "edi.document"
+
+    move_cancellation_ids = fields.One2many(
+        "edi.record.stock.cancellation",
+        "doc_id",
+        string="Cancelled Stock Moves",
+    )
+
+
+class EdiStockCancellationDocument(models.AbstractModel):
+    """Stock Cancellation document model
+
+    Stock Cancellation documents contain sets of moves
+    (grouped by sale & picking_type) which have been cancelled
+    since the previous EDI export. Note this report only makes sense
+    for outgoing picks linked to sales
```

Comment:
Note to self, this comment needs updating to reflect the fact this report now works for incoming/outgoing picks unrelated to sales